server/src/immich/api-v1/asset/asset.service.ts

```diff
@@ -105,6 +105,21 @@ export class AssetService {
     livePhotoFile?: UploadFile,
     sidecarFile?: UploadFile,
   ): Promise<AssetFileUploadResponseDto> {
+    if (dto.isReadOnly) {
+      // If this is read-only, we are likely importing. Ensure that the filesystem locations
+      // actually exist.
+      for (const fileToCheck of [file, livePhotoFile, sidecarFile]) {
+        if (fileToCheck) {
+          try {
+            await fs.access(file.originalPath, constants.R_OK);
+          } catch (error: any) {
+            this.logger.error(`Error importing file`, `${error}`);
+            throw new BadRequestException(`Error importing file`, `${error}`);
+          }
+        }
+      }
+    }
```

Comment:
Read only can be used independently of specifying an existing file that needs to be verified or not. Can you add a property to `UploadFile` that tracks if it is verified or not? Or if it is type upload or import? Then we can verify based on that property instead and keep it separate from RO functionality, which can be used in both situations.

I'd actually prefer a separate method here called importFile. You could put the import specific code there and then call uploadFile afterwards.

Reply:
Went with the last suggestion of having an importFile method which calls uploadFile afterwards, keeping the import logic separate.